addons/registry/2.7.1/install.sh

```diff
@@ -21,36 +43,39 @@ function registry() {
         render_yaml_file "$DIR/addons/registry/2.7.1/tmpl-persistentvolumeclaim.yaml" > "$DIR/kustomize/registry/persistentvolumeclaim.yaml"
         insert_resources "$DIR/kustomize/registry/kustomization.yaml" deployment-pvc.yaml
         insert_resources "$DIR/kustomize/registry/kustomization.yaml" persistentvolumeclaim.yaml
-
-        local will_migrate_pvc=
-        if ! registry_pvc_exists && object_store_exists && [ "$REGISTRY_USE_PVC_STORAGE" = "1" ]; then
-            will_migrate_pvc=1
-        fi
-        if object_store_exists && [ "$REGISTRY_USE_PVC_STORAGE" = "1" ]; then
-            logWarn "Registry migration in progres......"
-            render_yaml_file "$DIR/addons/registry/2.7.1/tmpl-configmap-migrate-s3.yaml" > "$DIR/kustomize/registry/configmap-migrate-s3.yaml"
-            insert_resources "$DIR/kustomize/registry/kustomization.yaml" configmap-migrate-s3.yaml
-            cp "$DIR/addons/registry/2.7.1/patch-deployment-migrate-s3.yaml" "$DIR/kustomize/registry/patch-deployment-migrate-s3.yaml"
-            insert_patches_strategic_merge "$DIR/kustomize/registry/kustomization.yaml" patch-deployment-migrate-s3.yaml
-        fi
     fi
 
-    kubectl apply -k "$DIR/kustomize/registry"
+    if [ "$WILL_MIGRATE_PVC" = "1" ]; then
+        logWarn "Registry migration in progres......"
 
-    if [ "$will_migrate_pvc" = "1" ]; then
-        logWarn "Registry will migrate from object store to pvc"
-        try_1m registry_pvc_migrated
-        logSuccess "Registry migration complete"
+        # Object store credentials already live in the previously created secret 
+        render_yaml_file "$DIR/addons/registry/2.7.1/tmpl-configmap-migrate-s3.yaml" > "$DIR/kustomize/registry/configmap-migrate-s3.yaml"
+        insert_resources "$DIR/kustomize/registry/kustomization.yaml" configmap-migrate-s3.yaml
+        cp "$DIR/addons/registry/2.7.1/patch-deployment-migrate-s3.yaml" "$DIR/kustomize/registry/patch-deployment-migrate-s3.yaml"
+        insert_patches_strategic_merge "$DIR/kustomize/registry/kustomization.yaml" patch-deployment-migrate-s3.yaml
     fi
+}
 
-    registry_cred_secrets
+export WILL_MIGRATE_PVC=
+function registry_will_migrate_pvc() {
+    if ! registry_pvc_exists && registry_object_store_exists && [ "$REGISTRY_USE_PVC_STORAGE" = "1" ]; then
+        WILL_MIGRATE_PVC=1            
```

Comment:
kinda a nit but can you refactor and take this global out and use the return code?